File: Assignment1/Mathutiltest2.cs
class Mathutiltest2{
       public static bool IsEven(int number)
       {
	       if(number%2==0)
	       {
		       return true;
	       }
	       else 
		       return false;
       }


     public static bool IsOdd(int number)
     {
	     if(number%2 !=0)
	     {
		     return true;
	     }
	     else
		     return false;
     }


      public static bool IsPrime(int number)
      {

	    if(number==1)
		    return false;
	    if(number==2 || number==3)
		    return true;
	    if((number%2)==0)
		    return false;
	    for(int i=3; i<=Math.Sqrt(number);i+=2)
	    {
		   if((number % i)==0)
			  return false;
	    }
	   return true;
    }


    public static int CountPrimes(int first, int last)
    {
	    int term,count=0;

	    for(term=first;term <= last; ++term)
	    {
		    if(IsPrime(term) == true)

			    count+=1;
	    }
	    return count;
    }

    public static int Reverse(int digit)
    {
	    int rev=0;
	    while (digit!=0){

		    rev =rev*10 + digit % 10;
		    digit/=10;
	    }
	    return rev;
    }

    public static int DigitCount(int num)
    {
	   int a=1;
	   int b=num;
	   int c=0;
	   do{
		   a=a*10;
		   c+=1;

	   }while(a<=b);
	   return c;
}
}

File: 2_JAVA/Assignment 2/MathUtiltest2.java

class MathUtiltest2{
	public static void main(String[] args){
	       int num = Integer.parseInt(args[0]);
	       int num1 = Integer.parseInt(args[1]);
	       int num2 = Integer.parseInt(args[2]);
	       int first = Integer.parseInt(args[3]);
	       int second = Integer.parseInt(args[4]);
	       int digit = Integer.parseInt(args[5]);
	       int count = Integer.parseInt(args[6]);
	       if(Mathutiltest1.isEven(num)==true){
		       System.out.printf("the number is even%n");
	       }
	       else
		       System.out.printf("the number is not even%n");
	

       
             if(Mathutiltest1.isOdd(num1)==true){
		    System.out.printf("the number is odd%n");
	     }
            else
	    System.out.printf("the number is not odd%n");


	    if(Mathutiltest1.isPrime(num2)==true){
		    System.out.printf("the number is prime%n");
	    }
	    else
		    System.out.printf("the number is not prime%n");


	    System.out.printf("the total prime numbers are %d%n",Mathutiltest1.countPrimes(first,second));
	    System.out.printf("the digit after reverse is  %d%n",Mathutiltest1.reverse(digit));
	    System.out.printf("the total numbers of digit is %d%n",Mathutiltest1.digitCount(count));
	}
}

